File: engine/model/src/main/scala/net/server/channel/packet/cash/use/UseMapleTvPacket.scala
package net.server.channel.packet.cash.use

class UseMapleTvPacket(private var _position: Short, private var _itemId: Int, private var _megaMessenger: Boolean, private var _ear: Boolean, private var _characterName: String, private var _messages: Array[String]) extends AbstractUseCashItemPacket(_position, _itemId) {
  def megaMessenger: Boolean = _megaMessenger

  def ear: Boolean = _ear

  def characterName: String = _characterName

  def messages: Array[String] = _messages
}

File: engine/engine-model/src/main/java/net/server/channel/packet/cash/use/UseMapleTvPacket.java
package net.server.channel.packet.cash.use;

public class UseMapleTvPacket extends AbstractUseCashItemPacket {
   private final Boolean megaMessenger;

   private final Boolean ear;

   private final String characterName;

   private final String[] messages;

   public UseMapleTvPacket(Short position, Integer itemId, Boolean megaMessenger, Boolean ear, String characterName, String[] messages) {
      super(position, itemId);
      this.megaMessenger = megaMessenger;
      this.ear = ear;
      this.characterName = characterName;
      this.messages = messages;
   }

   public Boolean megaMessenger() {
      return megaMessenger;
   }

   public Boolean ear() {
      return ear;
   }

   public String characterName() {
      return characterName;
   }

   public String[] messages() {
      return messages;
   }
}
